File: TriangularPentagonalHexagonal/TPH.py
#!/usr/bin/env python3
# All hexagonal numbers are triangular.

from math import sqrt

def is_pentagonal(n):
    x = (sqrt(24*n + 1) + 1)/6
    if (x).is_integer():
        return True
    else:
        return False

def hexagonal(n):
    return n*(2*n - 1)

n = 144

while not is_pentagonal(hexagonal(n)):
    n += 1

print(n, hexagonal(n))

File: TriangularPentagonalHexagonal/TPH.java
/*
Project Euler # 45

All hexagonal numbers are a subset of triangle numbers
*/

public class TPH {
    public static void main( String[] args ) {
        Long n = 144L;

        while ( !is_pentagonal(hexagonal(n)) )
            n++;

        System.out.println(n + ", " + hexagonal(n));
    }

    public static Long hexagonal( Long n ) {
        return n * (2*n - 1);
    }

    public static boolean is_pentagonal( Long n ) {
        double x = (Math.sqrt(24*n + 1) + 1)/6;

        if ( x == (int)x )
            return true;
        else
            return false;
    }
}
